browser/ui/views/playlist/playlist_action_dialogs.cc

```diff
@@ -118,10 +118,66 @@ class TiledItemsView : public views::BoxLayoutView {
 BEGIN_METADATA(TiledItemsView, views::BoxLayoutView)
 END_METADATA
 
+// A textfield that limits the maximum length of the input text.
+class BoundedTextfield : public views::Textfield {
+ public:
+  explicit BoundedTextfield(size_t max_length) : max_length_(max_length) {
+    length_label_ = AddChildView(std::make_unique<views::Label>());
+    length_label_->SetHorizontalAlignment(
+        gfx::HorizontalAlignment::ALIGN_RIGHT);
+    UpdateLengthLabel();
+  }
+  ~BoundedTextfield() override = default;
+
+  // views::Textfield:
+  void OnTextChanged() override {
+    Textfield::OnTextChanged();
+    UpdateLengthLabel();
+
+    // Double check the result as users can change contents via paste or
+    // composition.
+    // Note that this will be done in the next tick so that composition can
+    // finish its job.
+    base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+        FROM_HERE, base::BindOnce(&BoundedTextfield::TruncateText,
+                                  weak_ptr_factory_.GetWeakPtr()));
+  }
+
+  void InsertChar(const ui::KeyEvent& event) override {
+    if (GetText().size() >= max_length_) {
+      return;
+    }
+
+    Textfield::InsertChar(event);
+  }
+
+  void Layout() override {
+    views::Textfield::Layout();
+
+    length_label_->SetBoundsRect(GetContentsBounds());
+  }
+
+ private:
+  void TruncateText() {
+    if (auto text = GetText(); text.length() > max_length_) {
```

Comment:
Just a weird edge case, but what if you paste in some text which causes a Grapheme cluster to be cut in half? I guess it's probably okay to chop the grapheme in half?

Reply:
Wow, I didn't think about it. I thought the text.length should return the count of Grapheme, not the count of codes in the data, but it looks like the latter and could be chopped as you noted.
<img width="504" alt="image" src="https://github.com/brave/brave-core/assets/5474642/3890e28a-1032-478a-a210-2efc0d482f06">

 But I believe it should be okay, also as you noted, as each segment of Grapheme cluster is valid code, like this one https://emojipedia.org/family-man-woman-girl#technical ?

Reply:
Ugh, I got this - 
<img width="165" alt="image" src="https://github.com/brave/brave-core/assets/5474642/8704c4ac-a341-4eea-914c-b12a6042ef1e">

This emoji consists of Person Bowing, Zero Width Joiner and ♂️ Male Sign but in the screen shot above, the code `U+1F647 | U+200D | U+2642 U+FE0F` seems to be truncated to `U+1F647 | U+200D | U+2642`. I guess if we want to solve this perfectly, we need use ICU module but it feels too much for now. Let's see what happens.

Reply:
Yeah, I think it's probably so niche that it's not going to come up much :)